direct/src/showbase/ShowBase.py

```diff
@@ -197,69 +200,69 @@ def __init__(self, fStartDirect=True, windowType=None):
         self.requireWindow = ConfigVariableBool('require-window', True).value
 
         #: This is the main, or only window; see `winList` for a list of *all* windows.
-        self.win = None
-        self.frameRateMeter = None
-        self.sceneGraphAnalyzerMeter = None
+        self.win: Optional[GraphicsWindow] = None
```

Comment:
Actually, this may also be a GraphicsBuffer, in offscreen mode. But it should in any case be a GraphicsOutput.